### DISCUSSION TITLE
Lock file maintenance PR - external artifact update failure

### DISCUSSION THREAD

**Discussion by jej2003** [renovatebot/renovate#28257] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I am trying to understand the process associated with lock file maintenance PRs.  Renovate creates these PRs and updates all of the package-lock.json files in my project with what appear to be correct versions.  The PR always have an "Artifact Update Problem" with the following details:

```
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: project@1.0.0
npm ERR! notsup Not compatible with your version of node/npm: project@1.0.0
npm ERR! notsup Required: {"node":"20"}
npm ERR! notsup Actual:   {"npm":"9.8.1","node":"v18.18.2"}

npm ERR! A complete log of this run can be found in: /builds/renovate/renovate/cache/others/npm/_logs/2024-01-23T13_11_52_693Z-debug-0.log
```

While I understand the error, I believe I'm getting it because the version of node that renovate is running doesn't track with what the project requires (the renovate container is running 18.18.2 and the project requires 20).  I am guessing that I need to update the version of node renovate is running to fix this, is there any other solution?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by secustor** [comment, marked as answer]
If you are running inside ofva cotnainer you need to leave it set `install`

https://docs.renovatebot.com/self-hosted-configuration/#binarysource

**Reply by jej2003** [reply]
Even with setting this environment variable to install I am seeing the same behavior.

```
DEBUG: CLI config
       "config": {"logFileLevel": "debug", "token": "***********"}
DEBUG: Env config
       "config": {
         "hostRules": [],
         "repositoryCache": "true",
         "binarySource": "install",
         "baseDir": "/builds/renovate/renovate",
         "optimizeForDisabled": true
       }
DEBUG: Combined config
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": [
           "config:recommended",
           ":preserveSemverRanges",
           ":dependencyDashboard",
           ":rebaseStalePrs",
           ":enableVulnerabilityAlertsWithLabel('security')",
           "group:recommended"
         ],
         "endpoint": "https://hostname/api/v4/",
         "platform": "gitlab",
         "persistRepoData": true,
         "logFileLevel": "debug",
         "prConcurrentLimit": 10,
         "prHourlyLimit": 10,
         "onboardingConfig": {"extends": ["renovate/renovate-config"]},
         "repositories": ["repo"],
         "hostRules": [],
         "repositoryCache": "true",
         "binarySource": "install",
         "baseDir": "/builds/renovate/renovate",
         "optimizeForDisabled": true,
         "token": "***********"
       }
```

Logs
```
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: project@1.0.0
npm ERR! notsup Not compatible with your version of node/npm: project@1.0.0
npm ERR! notsup Required: {"node":"20"}
npm ERR! notsup Actual:   {"npm":"9.8.1","node":"v18.18.2"}

npm ERR! A complete log of this run can be found in: /builds/renovate/renovate/cache/others/npm/_logs/2024-04-08T02_04_54_503Z-debug-0.log
```

Is there something else that could be causing this?

**Reply by jej2003** [reply]
I think i found my issue.  The docker container I am using is based on the published renovate image, but it has a custom configuration for managers.  This is a new topic for me, but when I switched to the standard renovate docker image it did work as expected, so I'm assuming that the manager configuration is what was breaking my setup.
